server/src/main/java/org/elasticsearch/cluster/routing/allocation/DataTier.java

```diff
@@ -115,6 +130,15 @@ public static boolean isFrozenNode(final Set<DiscoveryNodeRole> roles) {
         return roles.contains(DiscoveryNodeRole.DATA_FROZEN_NODE_ROLE) || roles.contains(DiscoveryNodeRole.DATA_ROLE);
     }
 
+    public static List<String> parseTierList(String tiers) {
+        if (Strings.hasText(tiers) == false) {
```

Comment:
Bit weird that we treat all-whitespace as empty but otherwise we're whitespace-sensitive. (Acking that this is how it was before too, no action required)